Mono build fails with mono 5.18, requires C# 7.1

**Godot version:**
3.2 master (f2777c04)


**OS/device including version:**
Mageia 7 x86_64
msbuild 16.0 and:
```
mono --version
Mono JIT compiler version 5.18.1.0 (tarball Tue Apr  2 13:11:32 CEST 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug 
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       preemptive
        GC:            sgen (concurrent by default)
```

Same results on official buildsystem from @hpvb, still using the containers from 3.1.x with msbuild 15.0 and:
```
Mono JIT compiler version 5.18.1.3 (tarball Wed May 22 20:35:51 UTC 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug 
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       preemptive
        GC:            sgen (concurrent by default)
```

**Issue description:**
Building with `mono_glue=no` works fine to generate the glue, but the subsequent build with glue fails ~if `mono_static=yes` is used~ (I had a different error the first time with `mono_static=yes`, but now I always get this C# 7.1 error regardless of build options):

```
$ /usr/bin/scons LINKFLAGS=-fuse-ld=gold -j8 p=x11 warnings=extra werror=yes module_mono_enabled=yes mono_static=yes mono_prefix=/home/akien/Projects/godot/mono mono_glue=no
[...]
scons: done building targets.

$ ./bin/godot.x11.tools.64.mono --generate-mono-glue modules/mono/glue/
[...]
The solution for the Godot API was generated successfully

$ /usr/bin/scons LINKFLAGS=-fuse-ld=gold -j8 p=x11 warnings=extra werror=yes module_mono_enabled=yes mono_static=yes mono_prefix=/home/akien/Projects/godot/mono
[...]
CoreCompile:
  /usr/lib/mono/msbuild/15.0/bin/Roslyn/csc.exe /noconfig /nowarn:1701,1702 /nostdlib+ /errorreport:prompt /warn:4 /define:DEBUG /highentropyva+ /reference:/home/akien/Projects/godot/godot.git/bin/GodotSharp/Api/Debug/GodotSharp.dll /reference:/home/akien/Projects/godot/godot.git/bin/GodotSharp/Api/Debug/GodotSharpEditor.dll /reference:/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.BuildLogger/bin/Debug/GodotTools.BuildLogger.dll /reference:/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.Core/bin/Debug/GodotTools.Core.dll /reference:/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.IdeConnection/bin/Debug/GodotTools.IdeConnection.dll /reference:/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.ProjectEditor/bin/Debug/GodotTools.ProjectEditor.dll /reference:/usr/lib/mono/4.5-api/mscorlib.dll /reference:/usr/lib/mono/4.5-api/System.Core.dll /reference:/usr/lib/mono/4.5-api/System.dll /debug+ /debug:portable /optimize- /out:obj/Debug/GodotTools.dll /subsystemversion:6.00 /target:library /utf8output Build/MsBuildFinder.cs ExternalEditorId.cs Ides/GodotIdeManager.cs Ides/GodotIdeServer.cs Ides/MonoDevelop/EditorId.cs Ides/MonoDevelop/Instance.cs Internals/BindingsGenerator.cs Internals/EditorProgress.cs Internals/GodotSharpDirs.cs Internals/Internal.cs Internals/ScriptClassParser.cs Internals/Globals.cs Properties/AssemblyInfo.cs Build/BuildSystem.cs Utils/Directory.cs Utils/File.cs Utils/NotifyAwaiter.cs Utils/OS.cs GodotSharpEditor.cs BuildManager.cs HotReloadAssemblyWatcher.cs BuildInfo.cs BuildTab.cs BottomPanel.cs GodotSharpExport.cs CsProjOperations.cs Utils/CollectionExtensions.cs "/tmp/.NETFramework,Version=v4.5.AssemblyAttributes.cs"
Utils/NotifyAwaiter.cs(54,22): error CS8107: Feature 'default literal' is not available in C# 7.0. Please use language version 7.1 or greater. [/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj]
Done Building Project "/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj" (default targets) -- FAILED.
Done Building Project "/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.sln" (default targets) -- FAILED.

Build FAILED.

"/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.sln" (default target) (1) ->
"/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj" (default target) (4) ->
(CoreCompile target) -> 
  Utils/NotifyAwaiter.cs(54,22): error CS8107: Feature 'default literal' is not available in C# 7.0. Please use language version 7.1 or greater. [/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.16
scons: *** [bin/GodotSharp/Tools/GodotTools.dll] RuntimeError : 'msbuild' exited with error code: 1
Traceback (most recent call last):
  File "/home/akien/Projects/godot/godot.git/modules/mono/build_scripts/solution_builder.py", line 150, in run_command
    subprocess.check_call(args, env=env_override)
  File "/usr/lib64/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/msbuild', '/home/akien/Projects/godot/godot.git/modules/mono/editor/GodotTools/GodotTools.sln', '/p:Configuration=Debug']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/SCons/Action.py", line 1202, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "/home/akien/Projects/godot/godot.git/modules/mono/build_scripts/godot_tools_build.py", line 18, in build_godot_tools
    build_solution(env, solution_path, build_config)
  File "/home/akien/Projects/godot/godot.git/modules/mono/build_scripts/solution_builder.py", line 214, in build_solution
    run_command(msbuild_path, msbuild_args, env_override=msbuild_env, name='msbuild')
  File "/home/akien/Projects/godot/godot.git/modules/mono/build_scripts/solution_builder.py", line 152, in run_command
    raise RuntimeError("'%s' exited with error code: %s" % (name, e.returncode))
RuntimeError: 'msbuild' exited with error code: 1
scons: building terminated because of errors.
```

If the issue is that a more recent Mono version is needed for C# 7.1 support, the docs should be updated as they currently mention 5.12.0 or later.

For the reference, the second build step also generates 12 non-blocking warnings about the core API:
```
"/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp.sln" (default target) (1) ->
"/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj" (default target) (2) ->
(CoreCompile target) -> 
  ObjectType/ArrayMesh.cs(124,36): warning CS0108: 'ArrayMesh.BlendShapeMode' hides inherited member 'Mesh.BlendShapeMode'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/ArrayMesh.cs(39,21): warning CS0108: 'ArrayMesh.ArrayFormat' hides inherited member 'Mesh.ArrayFormat'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/ArrayMesh.cs(79,21): warning CS0108: 'ArrayMesh.ArrayType' hides inherited member 'Mesh.ArrayType'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/ARVRPositionalTracker.cs(68,39): warning CS0108: 'ARVRPositionalTracker.GetType()' hides inherited member 'object.GetType()'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/CPUParticles.cs(15,21): warning CS0108: 'CPUParticles.Flags' hides inherited member 'GeometryInstance.Flags'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/PathFollow2D.cs(94,21): warning CS0108: 'PathFollow2D.Rotate' hides inherited member 'Node2D.Rotate(float)'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/Slider.cs(90,38): warning CS0108: 'Slider.FocusMode' hides inherited member 'Control.FocusMode'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/LightOccluder2D.cs(36,20): warning CS0108: 'LightOccluder2D.LightMask' hides inherited member 'CanvasItem.LightMask'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/LineEdit.cs(216,38): warning CS0108: 'LineEdit.FocusMode' hides inherited member 'Control.FocusMode'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/ResourceFormatLoader.cs(12,52): warning CS1574: XML comment has cref attribute 'EditorImportPlugin' that could not be resolved [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
  ObjectType/Spatial.cs(168,102): warning CS1574: XML comment has cref attribute 'EditorSpatialGizmo' that could not be resolved [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]


"/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharp.sln" (default target) (1) ->
"/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharpEditor/GodotSharpEditor.csproj" (default target) (3) ->
  ObjectType/EditorSpatialGizmoPlugin.cs(73,31): warning CS0108: 'EditorSpatialGizmoPlugin.GetName()' hides inherited member 'Resource.GetName()'. Use the new keyword if hiding was intended. [/home/akien/Projects/godot/godot.git/modules/mono/glue/Managed/Generated/GodotSharpEditor/GodotSharpEditor.csproj]

    12 Warning(s)
    0 Error(s)
```

**Steps to reproduce:**
See above.

--- a/modules/mono/editor/GodotTools/GodotTools/Utils/NotifyAwaiter.cs
+++ b/modules/mono/editor/GodotTools/GodotTools/Utils/NotifyAwaiter.cs
@@ -51,7 +51,7 @@ namespace GodotTools.Utils
         {
             continuation = null;
             exception = null;
-            result = default;
+            result = default(T);
             IsCompleted = false;
             return this;
         }

